src/Features/CSharp/Portable/ConvertAutoPropertyToFullProperty/CSharpConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,226 @@
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.ConvertAutoPropertyToFullProperty;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Simplification;
+using static Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.SymbolSpecification;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertAutoPropertyToFullProperty
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(CSharpConvertAutoPropertyToFullPropertyCodeRefactoringProvider)), Shared]
+    internal class CSharpConvertAutoPropertyToFullPropertyCodeRefactoringProvider : AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider
+    {
+        internal override SyntaxNode GetProperty(SyntaxToken token)
+        {
+            var containingProperty = token.Parent.FirstAncestorOrSelf<PropertyDeclarationSyntax>();
+            if (containingProperty == null)
+            {
+                return null;
+            }
+
+            var start = containingProperty.AttributeLists.Count > 0
+                ? containingProperty.AttributeLists.Last().GetLastToken().GetNextToken().SpanStart
+                : containingProperty.SpanStart;
+
+            // Offer this refactoring anywhere in the signature of the property
+            var position = token.SpanStart;
+            if (position < start || position > containingProperty.Identifier.Span.End)
+            {
+                return null;
+            }
+
+            return containingProperty;
+        }
+
+        internal override async Task<string> GetFieldNameAsync(Document document, IPropertySymbol propertySymbol, CancellationToken cancellationToken)
+        {
+            var rules = await GetNamingRulesAsync(document, cancellationToken).ConfigureAwait(false);
+            return GenerateFieldName(propertySymbol, rules);
+        }
+
+        /// <summary>
+        /// Get the user-specified naming rules, then add standard default naming rules 
+        /// for both static and non-static fields.  The standard naming rules are added at the end 
+        /// so they will only be used if the user hasn't specified a preference.
+        /// </summary>
+        private static async Task<ImmutableArray<NamingRule>> GetNamingRulesAsync(
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            const string defaultStaticFieldPrefix = "s_";
+            const string defaultFieldPrefix = "_";
```

Comment:
i'm not sure we shoudl be making that choice for users.  Roslyn itself had no prefix on fields for tons of versions.  So, IMO, if there is no naming rule setup, then all we should do is lowercase the property name.  If that conflicts *then* add the _.  If tehre is a naming rule, we should respect it.